python/pyspark/sql/dataframe.py

```diff
@@ -470,7 +470,7 @@ def sampleBy(self, col, fractions, seed=None):
         +---+-----+
 
         """
-        if not isinstance(col, str):
+        if not isinstance(col, basestring):
```

Comment:
ah, got it. I just mean from SQL parser.

Similarly, as the unicode column name will be encoded by `name.encode('utf-8')`, it is now a `str` instance. In other words, the schema still stores column names as `str`. However, this change is allowing unicode input as `col`. I think there will be mismatching.